clock_control_sam_pmc.c missing <soc.h>

**Describe the bug**
The source at `zephyr/drivers/clock_control/clock_control_sam_pmc.c` is not buildable if `CONFIG_ARM_MPU` is not enabled. This is because it requires the `<soc.h>` header, which is included indirectly somewhere in in the header `<zephyr/arch/cpu.h>`, only when `CONFIG_ARM_MPU` which is not the desired behavior.

**To Reproduce**
Build the hello_world sample for the target board `sam4e_xpro`,
`west build -p -b sam4e_xpro zephyr/samples/hello_world`
disabling the config `CONFIG_ARM_MPU=n` in `zephyr/boards/arm/sam4e_xpro/sam4e_xpro_defconfig`



**Expected behavior**
The clock_control driver should build regardless of the ARM_MPU features inclusion.

**Impact**
The driver does not build unless ARM_MPU is enabled, which is unexpected since there is no direct link between the two components, which results in a very cryptic build error when building for the sam family of chips.

**Logs and console output**
```
/home/private/Documents/trackunit/zephyr_rtc/zephyr/drivers/clock_control/clock_control_sam_pmc.c: In function 'atmel_sam_clock_control_get_rate':
/home/private/Documents/trackunit/zephyr_rtc/zephyr/drivers/clock_control/clock_control_sam_pmc.c:89:25: error: 'SOC_ATMEL_SAM_MCK_FREQ_HZ' undeclared (first use in this function)
   89 |                 *rate = SOC_ATMEL_SAM_MCK_FREQ_HZ;
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~
```

**Environment (please complete the following information):**
 - OS: Linux Debian
 - Toolchain Zephyr SDK
 - Commit SHA efe9111b215f81ee3dbb48b3825a0b0f91cb6083


**Cross reference**
A fix for this bug is found in this PR: #59135

--- a/drivers/clock_control/clock_control_sam_pmc.c
+++ b/drivers/clock_control/clock_control_sam_pmc.c
@@ -13,6 +13,7 @@
 #include <zephyr/devicetree.h>
 #include <zephyr/drivers/clock_control.h>
 #include <zephyr/drivers/clock_control/atmel_sam_pmc.h>
+#include <soc.h>
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(clock_control, CONFIG_CLOCK_CONTROL_LOG_LEVEL);

